tests/phpunit/integration/Core/Modules/ModulesTest.php

```diff
@@ -62,6 +62,44 @@ function ( $instance ) {
 		);
 	}
 
+	public function test_get_available_modules__missing_dependency() {
+		$modules = new Modules( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );
+
+		add_filter(
+			'googlesitekit_available_modules',
+			function( $modules ) {
+				return array_filter(
+					$modules,
+					function( $module ) {
+						// Remove Analytics from the list of available modules.
+						return 'analytics' !== $module;
+					}
+				);
+			}
+		);
+
+		$available = array_map(
+			function ( $instance ) {
+				return get_class( $instance );
+			},
+			$modules->get_available_modules()
+		);
+
+		// Analytics is no longer present due to the filter above.
+		// Optimize is no longer present due to its dependency on Analytics.
+		$this->assertEqualSets(
+			array(
+				'adsense'            => 'Google\\Site_Kit\\Modules\\AdSense',
+				'analytics-4'        => 'Google\\Site_Kit\\Modules\\Analytics_4',
```

Comment:
I would expect `analytics-4` to also not be present if `analytics` were not available as the two are not fully independent. I don't think anything for GA4 would be shown without Analytics available to cause a problem but we should probably define `analytics` as a dependency of `analytics-4`.

What should happen if `analytics-4` was filtered out but not `analytics`? Can they depend on each other? 😄

Reply:
Did you mean by making use of `depends_on`, or by introducing a new approach for indicating module dependencies?

Making `analytics` a dependency of `analytics-4` via `depends_on` has the expected result of removing `analytics-4` when `analytics` is not available, so we could do that with minimal cost.

However, we can't make `analytics-4` a dependency of `analytics` via `depends_on`, as this results in the setup link for Analytics being disabled in normal usage, as `analytics-4` can't be setup independently:

![image](https://user-images.githubusercontent.com/18395600/186727985-33a5eb42-8c88-4dd2-b340-90c2de4fc7e8.png)

Reply:
> Making `analytics` a dependency of `analytics-4` via `depends_on` has the expected result of removing `analytics-4` when `analytics` is not available, so we could do that with minimal cost.

`analytics-4` isn't usable without `analytics` so I think this makes sense.

If someone filtered out GA4 though I think other problems would arise that probably aren't accounted for here but we can't add the dependency to `analytics` as that would be a cyclical dependency as well. I'm less worried about this scenario though as that would be "doing it wrong" I think. If anything, maybe we could make it so that `analytics-4` is always preserved so long as `analytics` is in the list of available modules (essentially preventing this possibility). It's a rather extreme edge case IMO though so, not a big deal, just thinking out loud.

Reply:
Cool, I've added `analytics` as a dependency of `analytics-4` via `depends_on`, so when filtering out `analytics`, `analytics-4` will be automatically removed too.

To be honest I don't think we should worry about the edge case of `analytics-4` being removed on its own. Analytics-4 is not exposed as an identifiable public facing module in its own right, and there's no actual reason to remove it in isolation - as you've said this would "doing it wrong", and on that basis I don't think we need to add logic for this scenario.